crates/revm/src/database.rs

```diff
@@ -4,15 +4,17 @@ use reth_provider::StateProvider;
 use revm::{
     db::{CacheDB, DatabaseRef},
     primitives::{AccountInfo, Bytecode},
+    Database,
 };
 
-/// SubState of database. Uses revm internal cache with binding to reth StateProvider trait.
-pub type SubState<DB> = CacheDB<State<DB>>;
+/// A REVM database that wraps a state provider and can store intra-block state in an in-memory
+/// cache.
+pub type SubState<SP> = CacheDB<StateProviderDB<SP>>;
```

Comment:
is this now only used in RPC and payload building?

tbh never really understood why it's named like that, the type alias makes things a bit more complex, suggest we either remove this entirely or rename this seperately.

Reply:
No, it's used in the executor too, so I left the name